docs/modules/json.md

```diff
@@ -0,0 +1,71 @@
+[< Back](.)
+
+json
+================
+
+The `json` module provides a simple interface to read and write [json files and strings](https://www.json.org/json-en.html).
+
+It contains the following classes:
+
+* [Json](#json)
+* [JsonOptions](#jsonoptions)
+* [JsonError](#jsonerror)
+
+## Json
+
+### Static Methods
+#### `static encode(object: Object): String`
+#### `static encode(object: Object, options:Num): String`
+Transform the object to a _Json_ encoded string. With default or custom options.
+
+#### `static decode(value:String, options:Num): Object`
+#### `static decode(value:String): Object`
+Returns a new _Json_ object with default or custom options.
+
+#### `static load(path:String): Object`
+#### `static load(path:String, options:Num): Object`
+Reads the contents of a file in `path` and returns a new _Json_ object with default or custom options.
+
+### `static save(path: String, object: Object)`
+### `static save(path: String, object:Object, options:Num)`
+This will encode the object and then save the result to a file specified in `path`. With default or custom options.
+
+## JsonOptions
+
+### Constants
+
+#### `static NIL: Num`
+_Default_ value. No options selected.
+
+#### `static ESCAPE_SLASHES: Num`
+This will encode solidus character (`/`). When converting a `Map` object to a `String`. This encoding is optional and is useful when you need to embed `JSON` inside `HTML <script>` tags. By default _DOME_ does not escape slashes.
+
+#### `static ABORT_ON_ERROR: Num`
+By default _DOME_ does not _Abort_ when there is a _JSON parsing error_. Use this option when you need to trigger a `Fiber.abort()` on parse error.
+
+### Example
+
+Use [Bitwise OR](https://wren.io/method-calls.html#operators) operator to select multiple options.
+
+```js
+Json.decode(myString, 
+	JsonOptions.ESCAPE_SLASHES |
+	JsonOptions.ABORT_ON_ERROR
+)
+```
+
+## JsonError
```

Comment:
It's not clear when this error is provided to the user, if at all.

Reply:
It would be provided if the _ABORT_ON_ERROR_ option is disabled.

```js
static decode(value, options) {
    var stream = JsonStream.new(value, options)
    stream.begin()

    var result = stream.result
    if (stream.error.found) {
      result = stream.error
    }

    stream.end()
    return result
  }
```

Reply:
I know that, but it's not stated in the documentation, so other people using these classes won't be able to tell.

Reply:
added: 

This object is returned on calls to _Json.decode_ only if the  `JsonOptions.ABORT_ON_ERROR` default behaviour is disabled and a parse error was found.

to the JsonError docs